### DISCUSSION TITLE
Next Patch (v4.4.1)

### DISCUSSION THREAD

**Discussion by Siemienik** [exceljs/exceljs#2573] [General]
Hello everyone! I am kickstarting this discussion to pinpoint the key issues and bugs that need addressing in the upcoming patch release 4.4.1. Your insights on the PRs and Issues that should take precedence are highly valued. This support will enable me to allocate more time to this project. Engage in the discussion here or on our Discord.

For me, the most pressing concerns at the moment are:

* #683 - I've observed that it's triggering CI errors like: [CI Error Link](https://github.com/exceljs/exceljs/actions/runs/6266446815/job/17017455540)
* #514

Some other issues, although not guaranteed for inclusion in 4.4.1, will be considered based on my availability or contributions from others:
* #1681
* #1904
* #1399
* #2565

Proposed PRs:
* #2559
* #2571
* #2566
* #2562 - Awaiting resolution of discussion, thereafter it will be either merged or declined.

Feel free to reach out to me at consulting@siemienik.com for further discussion or sponsorship queries.

Looking forward to your active participation and collaboration!

---
[![SiemaTeam](https://discordapp.com/api/guilds/976854442009825321/widget.png?style=banner2)](https://discord.gg/siema)

**Comment by Siemienik** [comment]
One bug, actually a regression which should be fixed ASAP:
https://discord.com/channels/976854442009825321/1029874274804179024/1167100475871608912


> ~~ By Balak: 
> 
> hello, I am seeing an issue with 4.4.0 which goes away if I revert back to 4.3.0 - I don't see anything in the changelog about this. I have a spreadsheet with images in the last cell of each row.  With 4.40, the first image is squished and extra row and column are being created. I see the extra erroneous column created in the debugger after addImage on the worksheet is called. Thanks

**Reply by testsunhub** [reply]
hello,How do I get font size on a cell that has never been styled?

**Comment by Siemienik** [comment]
* https://github.com/exceljs/exceljs/pull/1222 / https://github.com/exceljs/exceljs/pull/1345 / https://github.com/exceljs/exceljs/pull/1938

**Comment by Siemienik** [comment]
* #2575 - with experimental flag

**Comment by Siemienik** [comment]
https://discord.com/channels/976854442009825321/1097805766339088424/1168977388370149376

**Comment by hubyhub** [comment]
hello Siemienik!
I am not quite sure, whether it is still an open issue or I am missing something. because the issue is flagged as "completed".

But I still experience the same performance problems as mentioned here:  
[Heavily degrading performance in cell merges #637](https://github.com/exceljs/exceljs/issues/637)
My table has several thousand rows and around 20 columns and I need to merge the cells of some of those columns across multiple rows. 

Without the check,  the excel is built successfully within seconds.

is it possible to just avoid this check via a flag as mentioned by [unpollito](https://github.com/unpollito)?
 
I am using  Exceljs v4.4.0

**Reply by hubyhub** [reply]
Ah I have just seen that there is already a pull request  
See here: [fix: inefficient merge check for large amount of merged cells within](https://github.com/exceljs/exceljs/pull/2691)

**Comment by naresh5265** [comment]
@Siemienik  Any plans to address this open issue https://github.com/exceljs/exceljs/issues/2715 in Next Patch (v4.4.1)

**Comment by aorsten** [comment]
@Siemienik Thanks for the great work here. I notice there's a while since your last activity - is this repo a priority these days? 

I'd like to throw in a suggestion for adding this to the patch: https://github.com/exceljs/exceljs/pull/2691

**Comment by Arvid-L** [comment]
If there will be any activity on this package in the future I'd really appreciate if this PR could be added to this patch (or minor) version, depending where it fits better:

https://github.com/exceljs/exceljs/pull/2736

**Comment by TheAsda** [comment]
I hope this package is not dead. We are using it in production and we have a serious bug that affects streaming excel files while generating. So here is the PR to fix the issue #2803

**Reply by Catscratch** [reply]
Is this project still maintained? I would ask to merge #2876 please.

**Comment by protobi-pieter** [comment]
Branch https://www.npmjs.com/package/@protobi/exceljs includes:
- #2876 Fix issue adding multiple images
- #2803 Fix corruption bug
- #2736 Conditional formatting improvements

**Comment by protobi-pieter** [comment]
@Siemienik  Here's our review of open pull requests on the master repo, classified by complexity, completeness, salience, and type, and prioritized. https://github.com/protobi/exceljs/wiki/Analysis-of-open-pull-requests-in-ExcelJS    I think we're at a stable place for our internal uses cases, so we can test this in practice for a while (weeks, months) and aim for a composite pull request.
